cvat-core/src/frames.ts

```diff
@@ -555,7 +564,27 @@ export function getFramesMeta(type: 'job' | 'task', id: number, forceReload = fa
                     ...serialized,
                     deleted_frames: Object.fromEntries(serialized.deleted_frames.map((_frame) => [_frame, true])),
                 });
-                resolve(framesMetaData);
+                // When we get new framesMetaData from server there can be some unsaved data
+                // here we merge new meta data with cached one
+                if (previousCache instanceof Promise) {
+                    previousCache.then((prevMeta) => {
+                        const updatedFields = prevMeta.getUpdated();
+                        for (const key in updatedFields) {
+                            if (Object.hasOwn(updatedFields, key) && key.startsWith('deletedFrames')) {
```

Comment:
Why not `key === 'deletedFrames'`?

Reply:
Makes sense. Moved all logic with keys into FramesMetaData class, now we will just use `.deletedFrames` prop from `.getUpdated()`